apps/solution/src/components/people/people.tsx

```diff
@@ -1,18 +1,47 @@
 import { Grid } from '@libs/design';
+import { Dispatch, SetStateAction, useState } from 'react';
 import Person from '../person/person';
 import usePeople from '../person/usePeople';
 
 import style from './people.module.scss';
 
+function normalize(text: string) {
+  return text.toLowerCase().normalize('NFD').replace(/[\u0300-\u036f]/g, "")
+}
+
 export function People() {
   const people = usePeople();
+  const [searchTerm, setSearchTerm] = useState("")
+
+  const filteredPeople = people?.filter(person => {
+    const searchTermNormalized: string = normalize(searchTerm)
+    return normalize(person.lastname).includes(searchTermNormalized) || normalize(person.firstname).includes(searchTermNormalized)
+  })
 
   return (
-    <Grid className={style.container}>
-      {people?.map((person) => <Person key={person.id} person={person} />) ||
-        'Loading ...'}
-    </Grid>
+    <>
+      <SearchBar searchTerm={searchTerm} setSearchTerm={setSearchTerm} />
+      <Grid className={style.container}>
+        {filteredPeople?.map((person) => <Person key={person.id} person={person} />) ||
+          'Loading ...'}
+      </Grid>
+    </>
   );
 }
 
+
+export interface SearchBarProps {
+  searchTerm: string;
+  setSearchTerm: Dispatch<SetStateAction<string>>;
```

Comment:
I wonder if this is the right typing ? 
I feel this is too specific. I would rather prefer a more generic function definition, which works no matter if the passed prop is  a Dispatcher from `useState` or another function for any other behavior.
What do you think about it ?

Reply:
Function seem to be a banned type though

```
Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape.eslint[@typescript-eslint/ban-types](https://typescript-eslint.io/rules/ban-types)
```

Reply:
I guess you tried to type using `Function`, right ? 
Maybe you can use  something like `(value: string) => void` instead, which is generic enough to be used by any parent component, and still correctly type both function's parameters and results.
What do you think ?

Reply:
I didn't know you could just pass these kind of types there 🤔, this allows us to limit it to string in input so I agree with you.